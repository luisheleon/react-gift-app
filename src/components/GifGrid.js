import React from "react";
import PropTypes from "prop-types";
import CardImage from "./CardImage";
import { Grid } from "@mui/material";
import useGetGifts from "../hooks/useGetGifts";

function GifGrid({ category }) {
  const { data: images, loading } = useGetGifts(category);

  return (
    <React.Fragment>
      {loading && "Cargando... "}
      <h3>Categories {category} </h3>
      <Grid container spacing={2}>
        {images.map((img) => (
          <Grid item md={2} key={`grid-${img.id}`}>
            <CardImage {...img} key={img.id} category={category} />
          </Grid>
        ))}
      </Grid>
    </React.Fragment>
  );
}

GifGrid.propTypes = {
  category: PropTypes.string,
};

export default GifGrid;
