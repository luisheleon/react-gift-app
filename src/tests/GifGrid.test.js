import GifGrid from "../components/GifGrid";
import { shallow } from "enzyme";

describe("Test <GifGrid />", () => {
  let wrapper;
  let category = "Ducks";

  beforeEach(() => {
    wrapper = shallow(<GifGrid category={category} />);
  });

  test("should verify snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
