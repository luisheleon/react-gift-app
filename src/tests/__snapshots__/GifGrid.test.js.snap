// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGrid /> should verify snapshot 1`] = `
<Fragment>
  Cargando... 
  <h3>
    Categories 
    Ducks
     
  </h3>
  <ForwardRef(Grid)
    container={true}
    spacing={2}
  />
</Fragment>
`;
